### PR TITLE
CI preparations

### PR DESCRIPTION
**Pull request by Vittly**
Addressed to https://github.com/bem/webpack-bem-plugin/issues/11 and https://github.com/bem/webpack-bem-plugin/issues/3

### CODE REVIEW

**Comment by Vittly** [comment]
`BemRequest` and other helpers should be covered under https://github.com/bem/webpack-bem-plugin/issues/10

`index.js`, `BemModuleFactoryPlugin` and `BemDependency` are more about webpack internals. I've not decided about writing unit test for them. May be e2e is enough

**Comment by Vittly** [comment]
@zxqfox @Yeti-or and @awinogradov can some of you check?

**Comment by Vittly** [comment]
I'll wait till sunday and merge. Please check if you can

File: test/unit/helpers/importedFiles.test.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+jest.mock('../../../src/helpers/namingConvention',
+    () => ({ getFromLevel: () => '[origin]' })
+);
+
+jest.mock('../../../src/helpers/cellPath',
+    () => (cell, naming) => {
+        expect(naming).toBe('[origin]');
+
+        return `${cell.layer}.blocks/${cell.entity.block}/${cell.entity.block}.${cell.tech}`;
```
Comment (qfox) [inline test/unit/helpers/importedFiles.test.js:17]: `file.naming.stringify` doesnt work well?

File: test/unit/generator.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            'require(\'dir/bar.css\'),' + '\n' +
+            'require(\'dir/foo.js\'),' + '\n' +
+            '(require(\'dir/bar.js\').default || require(\'dir/bar.js\')).applyDecls()'
+        );
+    });
+
+    it('should filter nonexistent files', () => {
+        fs.existsSync.mockImplementation(path => {
+            switch (path) {
+                case 'dir/foo.js':
```
Comment (qfox) [inline test/unit/generator.test.js]: Не хочешь оформить в виде hash?
`const inMemoryFs = { dir: { 'foo.js': 'qwe' } }`, и возвращать `_.flat(inMemoryFs).hasOwnKey(path)`

Ну или хотя бы `return {'dir/foo.js': true, 'dir/foo.js': false}[path];`

Switch выглядит громостко

**Review by qfox — APPROVED** [review]

File: test/unit/helpers/importedFiles.test.js (hunk 1 above)
Comment (Vittly) [reply test/unit/helpers/importedFiles.test.js:17]: I also mocked getter of naming (see few lines before - naming is simple string) so true `bem/sdk.naming.*.stringify` will fail.

Plugin doesn't use `bem/sdk.*` directly only through "helpers". After https://github.com/bem/webpack-bem-plugin/issues/10 we will remove helpers and such mocks won't be needed.

On the other side mocking of all dependencies is good for unit testing. Whole thing is tested by e2e

File: test/unit/generator.test.js (hunk 2 above)
Comment (Vittly) [reply test/unit/generator.test.js]: I like second:)

Comment (Vittly) [reply test/unit/generator.test.js]: For now `fs.existsSync.mockImplementation(path => path === 'dir/foo.js');` is the smallest

**Pull request merged**